jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/editor/CodeWhispererEditorUtil.kt

```diff
@@ -36,24 +37,22 @@ object CodeWhispererEditorUtil {
         return FileContextInfo(caretContext, fileName, programmingLanguage)
     }
 
-    fun ProgrammingLanguage.toCodeWhispererLanguage() = when (languageName) {
-        CodewhispererLanguage.Python.toString() -> CodewhispererLanguage.Python
-        CodewhispererLanguage.Java.toString() -> CodewhispererLanguage.Java
-        CodewhispererLanguage.Javascript.toString() -> CodewhispererLanguage.Javascript
-        "plain_text" -> CodewhispererLanguage.Plaintext
-        else -> CodewhispererLanguage.Unknown
-    }
-
     val PsiFile.programmingLanguage: ProgrammingLanguage
-        get() = ProgrammingLanguage(lowerCase(this.fileType.name))
+        get() {
+            val fileTypeName = lowerCase(this.fileType.name)
+            return when {
+                fileTypeName.contains("jsx") -> ProgrammingLanguage(CodewhispererLanguage.Javascript)
+                else -> ProgrammingLanguage(fileTypeName)
+            }
+        }
```

Comment:
Do we want to add a new value for jsx in telemetry?
For the final mapping when we are creating the request we can map `jsx harmony` to `javascript`, but for other purposes (e.g. sending telemetry events) we can get events of .jsx files if we want.

Reply:
Also -- consider moving the mapping helper into CodeWhispererLanguageManager -- looks cleaner IMO.

Reply:
> Do we want to add a new value for jsx in telemetry? For the final mapping when we are creating the request we can map `jsx harmony` to `javascript`, but for other purposes (e.g. sending telemetry events) we can get events of .jsx files if we want.


Yea sounds good, might need another type of enum CodeWhispererLanguage to represent JSX. Will open another PRs update definition + plugin telemetry sending.

Reply:
> Also -- consider moving the mapping helper into CodeWhispererLanguageManager -- looks cleaner IMO.

updated!